error-curation.md

```diff
@@ -0,0 +1,87 @@
+* We are calling these "rules" not "errors"
+  * They may be found eagerly, may just be disable-able warnings, or may even be just best practice rules
+* Rules can be for many things:
+  * Errors (e.g. SDK errors, static analysis errors, server errors, etc)
+  * Guidelines (e.g. best practices, warnings, etc)
+* Every rule has an identifier in the form of `TMPRLXXXX`
+  * "Categories" of rules are given a 100 block as rarely is more needed
+    * Category 100 blocks should try not to be next to another so space can be given in some rare case we exceed 100 in
+      a category
+    * There are no rules on what a category is, just a grouping of like things
+  * The rules within a category do not have to be ordered in any manner. Usually you just use the next available.
+  * `0` is never used in the first two digits
+  * Can discuss the first thousands-block digit though initial thought for first two is:
+    * `TMPRL1XXX` block is for SDK-specific and workflow/activity authoring things
+    * `TMPRL2XXX` block is for server-specific and operation things
+* Rule/error messages across Temporal should be prepended with `"[TMPRLXXXX] "` (sans quotes)
+  * Note the brackets and space after.
+  * We are intentionally not providing a stricter requirement than this. There is intentionally no requirement to make
+    these available for programmatic consumption and we are not even going to guarantee these are even present or there
+    is a stable URL to learn more.
+  * SDK, gRPC API and other programmatic places where errors occur do not need to go out of their way to separate this
+    from the message string. There is limited value. From POV of code, this might as well be the beginning of an English
+    sentence and no more. Forcing stricter format/abstraction is more cost than benefit.
+* All rules are housed in a single repository
+  * Maybe something like https://github.com/temporalio/temporal-rules https://github.com/temporalio/rule-library
+  * `rules/README.md` will be the primary index of all categories
+  * `rules/TMPRL11XX.md` with the literal `XX.md` suffix but the two-digit category will be the index of all rules in
+    that category
+  * `rules/TMPRL1234.md` will be the literal page
+    * Consideration was given to breaking these out into subdirectories, but experience has shown a nice, common
+      `https://github.com/temporalio/temporal-rules/blob/main/rules/<RULE_NUM>.md` URL for _all_ rules has a lot of
+      benefits
+  * This project does not concern itself with the eventual promotion of this content to regular docs, just that it needs
+    to be in a repository for review/submission by Temporal developers
+* The `rules/TMPRL1234.md` file will contain rule details
+  * No strict metadata/frontmatter required at this time
+  * Ideally the rules should have at least the following headings:
+    * `# <TMPRL1234> - <one-sentence-description>`
+    * `## Cause`
+    * `## Description`
```

Comment:
wonder if usually the cause has ~all the description?

Reply:
Yeah, I could see that for simple stuff. I mention this is not a strict guideline for things like this. The reason for description separate is to separate "why" from "where". But can be combined if simple enough, sure.